addons/crm/models/crm_lead.py

```diff
@@ -1307,6 +1323,60 @@ def _handle_salesmen_assignment(self, user_ids=False, team_id=False):
     # CLASSIFICATION TOOLS
     # --------------------------------------------------
 
+    @api.depends('email_from', 'partner_id', 'contact_name', 'partner_name')
+    def _compute_potential_lead_duplicates(self):
+
+        MIN_EMAIL_LENGTH = 7
+        MIN_NAME_LENGTH = 6
+        SEARCH_RESULT_LIMIT = 21
+
+        def return_if_relevant(model_name, domain):
+            # include archived records and transcend multi-company record rules
+            model = self.env[model_name].sudo().with_context(active_test=False)
+            res = model.search(domain, limit=SEARCH_RESULT_LIMIT)
+            return res if len(res) < SEARCH_RESULT_LIMIT else model
+
+        def get_email_to_search(email):
+            """ Return email domain or full email if in most common domains list """
+            if not email or len(email) < MIN_EMAIL_LENGTH:
+                return False
+            parts = email.rsplit('@', maxsplit=1)
+            if len(parts) > 1:
+                email_domain = parts[1]
+                if email_domain not in iap_tools._MAIL_DOMAIN_BLACKLIST:
+                    return '@' + email_domain
+            return email
+
+        for lead in self:
+
+            lead_id = lead._origin.id if isinstance(lead.id, models.NewId) else lead.id
+            common_lead_domain = [
+                ('id', 'not in', [lead_id])
+            ]
+
+            duplicate_lead_ids = self.env['crm.lead']
```

Comment:
This recordset is using current user environment but return_if_relevant returns a sudo-ed record set. This is a bit misleading as people will not know it. Do you really require the sudo() in return_if_relevant ?